DatadogRUM/Sources/Instrumentation/Views/SwiftUI/SwiftUIViewNameExtractor.swift

```diff
@@ -0,0 +1,159 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import UIKit
+import DatadogInternal
+
+// MARK: - SwiftUIViewNameExtractor
+/// Protocol defining interface for extracting view names for SwiftUI views
+internal protocol SwiftUIViewNameExtractor {
+    func extractName(from: UIViewController) -> String?
+}
+
+// MARK: - SwiftUIReflectionBasedViewNameExtractor
+/// Default implementation that extracts SwiftUI view names using reflection and string parsing
+internal struct SwiftUIReflectionBasedViewNameExtractor: SwiftUIViewNameExtractor {
+    private let createReflector: (Any) -> TopLevelReflector
+
+    init(
+        reflectorFactory: @escaping (Any) -> TopLevelReflector = { subject in
+            Reflector(
+                subject: subject,
+                // MARK: TODO: RUM-9035 - Telemetry
+                telemetry: NOPTelemetry()
+            )
+        }
+    ) {
+        self.createReflector = reflectorFactory
+    }
```

Comment:
/nit It would read better if we have a `convenience init` that take a `telemetry` parameter.

Reply:
Sorry my comment was not clear, what I meant was to remove the default `reflectorFactory` value and add a `convenience init(telemetry: Telemetry = NOPTelemetry())` :)